Content.Server/Administration/Systems/BwoinkSystem.cs

```diff
@@ -107,14 +124,129 @@ private void OnOverrideChanged(string obj)
             _overrideClientName = obj;
         }
 
-        private void OnPlayerStatusChanged(object? sender, SessionStatusEventArgs e)
+        private async void OnPlayerStatusChanged(object? sender, SessionStatusEventArgs e)
         {
+            if (e.NewStatus == SessionStatus.Disconnected)
+            {
+                if (_activeConversations.TryGetValue(e.Session.UserId, out var lastMessageTime))
+                {
```

Comment:
Shouldn't you use a canceltoken for this? E.g. if they connect, disconnect, then reconnect before DB response I feel this is going to break.

Reply:
I'm going to pretend DB stuff gets returned in order and hope no one notices this comment in 2 years.